uber/site_sections/jobs.py

```diff
@@ -45,100 +53,51 @@ def index(self, session, location=None, message=''):
             'location':  location,
             'setup':     [j for j in jobs if j.type == c.SETUP],
             'teardown':  [j for j in jobs if j.type == c.TEARDOWN],
-            'checklist': session.checklist_status('creating_shifts', location),
+            'checklist': location and session.checklist_status('creating_shifts', location),
             'times':     [(t, t + timedelta(hours=1), by_start[t]) for i, t in enumerate(times)]
         }
 
     def signups(self, session, location=None, message=''):
-        if location is None:
+        if not location:
             location = cherrypy.session.get('prev_location') or c.JOB_LOCATION_OPTS[0][0]
+        location = None if location == 'All' else location
         cherrypy.session['prev_location'] = location
 
-        jobs, shifts, attendees = session.everything(location)
         return {
             'message':   message,
             'location':  location,
-            'jobs':      jobs,
-            'shifts':    Shift.dump(shifts),
-            'checklist': session.checklist_status('postcon_hours', location)
+            'attendees': session.staffers_for_dropdown(),
+            'jobs':      [job_dict(job) for job in session.jobs(location)],
+            'checklist': location and session.checklist_status('postcon_hours', location)
         }
 
     def everywhere(self, session, message='', show_restricted=''):
-        shifts = defaultdict(list)
-        for shift in session.query(Shift).options(joinedload(Shift.attendee)).all():
-            shifts[shift.job_id].append(shift_dict(shift))
         return {
             'message': message,
             'show_restricted': show_restricted,
-            'attendees': [{
-                'id': id,
-                'full_name': full_name.title()
-            } for id, full_name in session.query(Attendee.id, Attendee.full_name)
-                                          .filter_by(staffing=True)
-                                          .order_by(Attendee.full_name).all()],
-            'jobs': [job_dict(job, shifts[job.id])
-                     for job in session.query(Job)
-                                       .filter(Job.start_time > localized_now() - timedelta(hours=2))
-                                       .filter_by(**({} if show_restricted else {'restricted': False}))
-                                       .order_by(Job.start_time, Job.location).all()]
+            'attendees': session.staffers_for_dropdown(),
+            'jobs': [job_dict(job) for job in session.jobs()
+                                                     .filter(Job.start_time > localized_now() - timedelta(hours=2))
+                                                     .filter_by(**{} if show_restricted else {'restricted': False})]
         }
 
-    @ajax
-    def assign_from_everywhere(self, session, job_id, staffer_id):
```

Comment:
Previously we had two different pages which showed the same type of lists of shifts.  One was the "all untaken shifts" page which was coded with efficiency in mind and one was the "all shifts for a department" which was not.  I refactored the latter to use the same approach as the former, which meant I could consolidate these backend methods instead of maintaining two separate versions.

This is one case where the better performance actually made the code even cleaner, which is nice.